aiida/engine/processes/calcjobs/calcjob.py

```diff
@@ -337,8 +337,12 @@ def parse_scheduler_output(self, retrieved):
         filename_stdout = self.node.get_option('scheduler_stdout')
 
         detailed_job_info = self.node.get_detailed_job_info()
+
         if detailed_job_info is None:
-            self.logger.warning('could not parse scheduler output: the `detailed_job_info` attribute is missing')
+            self.logger.info('could not parse scheduler output: the `detailed_job_info` attribute is missing')
+        elif detailed_job_info.get('retval', 0) != 0:
+            self.logger.info('could not parse scheduler output: return value of `detailed_job_info` is non-zero')
+            detailed_job_info = None
```

Comment:
> Isn't this a breaking change? Are we sure there is no other place in the code that will check the `retval` and have a different behaviour if its `None` or a non zero integer?

This can't be breaking because the code that is changed hasn't been released yet.

> Also, isn't a value other than 0 a parse that actually had a problem, and thus worthy of a `warning` type of log?

As described in the commit message, the command can fail just because the scheduler installation does not support it. This is for example the case for vanilla SLURM schedulers. Since retrieving and parsing the detailed job info is completely optional, I don't think it should be a warning if it fails. Ideally, we could detect the special case of it failing for SLURM just not having accounting enabled, but we can't in the generic way, and there may be comparable cases for other schedulers.

Reply:
Ahh, ok, I missunderstood that last paragraph. I said nothing then.